src/lib/db.js

```diff
@@ -59,7 +64,11 @@ function getObservationTimestampStream (options, cb) {
     done = cb
     opts = options
   }
-  observationsTimestampIndex.ready(() => done(null, observationsTimestampIndex.getDocumentStream(opts)))
+  if (timestampIndexReady) {
```

Comment:
I think you can safely revert this commit: `ready` returns when the index is ready, which is instantaneously if the indexes are already prepared.